core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -229,6 +229,23 @@ private[spark] class Executor(
     ManagementFactory.getGarbageCollectorMXBeans.asScala.map(_.getCollectionTime).sum
   }
 
+  /**
+   * Only in local mode, we have to prevent the driver from setting the active SparkSession
+   * in the executor threads.  See SPARK-23894.
+   */
+  private lazy val clearActiveSparkSessionMethod = if (Utils.isLocalMaster(conf)) {
```

Comment:
Do we still need to call this if we can resolve the root cause in https://github.com/apache/spark/pull/21190, and issue an error promptly like what you said in https://github.com/apache/spark/pull/21185#issuecomment-385996077?

Reply:
I've added this check in https://github.com/apache/spark/pull/21190

Reply:
agreed, I closed this PR